Fix incorrect EBB

qwe661234 opened the pull request:
1. When block cache is full, we need to clear the EBB information if the block has been replaced.
2. BB should not be extended if the exception hanlder is invoked.
3. Based on running benchmark SciMark2, frequently compilation cause error in MIR, so the threashold should be adjusted to pass the benchmark.

jserv commented on src/cache.c:
```diff
@@ -20,7 +20,7 @@
  * exceeds the THRESHOLD, the JIT compiler flow is triggered.
  */
-#define THRESHOLD 1000
+#define THRESHOLD 100000
```
Do you employ any scientific methods or criteria to determine or adjust the value of the threshold?

qwe661234 replied:
I just simply increase threshold to the value that can pass the Sci Mark2.

jserv replied:
> I just simply increase threshold to the value that can pass the Sci Mark2.

Additionally, it is important to provide a clear explanation of the reasoning behind such adjustments and reference the intended target (SciMark2) accordingly.

Always refer to it as "[SciMark2](https://math.nist.gov/scimark2/)" as it ought to be.

jserv commented on src/rv32_template.c:
```diff
@@ -56,25 +56,55 @@ RVOP(jalr, {
-        return ir->branch_taken->impl(rv, ir->branch_taken);     \
-    }                                                            \
+#if RV32_HAS(JIT)
+#define BRANCH_FUNC(type, cond)                                   \
```
Can you shorten the definition of macro `BRANCH_FUNC`?

qwe661234 committed:
Fix incorrect EBB

1. When block cache is full, we need to clear the EBB information if the
block has been replaced.
2. BB should not be extended if the exception hanlder is invoked.
3. Based on running benchmark SciMark2, frequently compilation cause error
in MIR, so the threashold should be adjusted to pass the benchmark.
